### PR TITLE
FEATURE: Allow to skip validation of uninitialized proxies

### PR DESCRIPTION
**Pull request by kdambekalns**
The new option `skipUnInitializedProxies` allows to skip validation
of uninitialized proxy instances, to avoid recursions down into
unchanged hierarchies.

### CODE REVIEW

**Comment by kitsunet** [comment]
The interesting question is, when do we actually ever need to validate uninitialised proxies. Because this must be an existing entity that was not changed, which means it went through validation at the time it was written last...

**Comment by albe** [comment]
As said, theoretically, you could have an aggregate with an entity inside, that is "valid" depending on some external (but still within the aggregate) constraints/state. So even if that entity itself wasn't changed and hence not initialized, the validation is important, because that other state could have changed. I have a hard time imagining a practical example straight away, but it's a possibility.

Basically: you should not partially validate an aggregate - the aggregate is to be considered a single whole. And as long as you build sane (small) Aggregates, that is not an issue.

**Comment by kdambekalns** [comment]
But in that case you'd write a custom validator for said object, no?

**Comment by albe** [comment]
Maybe, but if the validator is for that sub entity object (and not the root entity), it wouldn't be invoked any longer and still lead to partially (in)valid aggregates.

**Comment by kdambekalns** [comment]
I'd say in that case you should write a validator for that root entity. I still think this is a valid case, and I wrote this to fix a real problem a customer had. How do we proceed?

**Comment by albe** [comment]
Wait, still needs proper documentation as it's a new feature.

**Comment by kdambekalns** [comment]
@albe Added some documentation, is that sufficient?

**Comment by albe** [comment]
Looks good! Thanks :)
